### PR TITLE
Feature for enabling instructors to rename saved queries in email widget

### PR DESCRIPTION
**Pull request by caijim**
Feature for enabling instructors to rename saved queries in email widget.

groupname

emailsync

@kluo

### CODE REVIEW

File: lms/djangoapps/instructor/tests/test_api.py [hunk 1]
```diff
@@ -422,9 +424,48 @@ def test_saving_queries(self):
         saved_queries = self._get_saved_queries()
         self.assertEquals(len(saved_queries), 2)
         query_group = saved_queries[0]['group']
+        query_title = saved_queries[0]['group_title']
+        self.assertEquals(query_title, '')
+        #save another set of queries
```
Comment (kluo) [inline lms/djangoapps/instructor/tests/test_api.py]: space after `#`

File: lms/djangoapps/instructor/tests/test_api.py [hunk 2]
```diff
@@ -422,9 +424,48 @@ def test_saving_queries(self):
...
+        self._save_query(temp_ids)
+        new_group_name = "new_name"
+        #rename the query
```
Comment (kluo) [inline lms/djangoapps/instructor/tests/test_api.py]: space after `#`

File: lms/djangoapps/instructor/tests/test_api.py [hunk 3]
```diff
@@ -422,9 +424,48 @@ def test_saving_queries(self):
...
+        rename_url = reverse('save_group_name', kwargs={'course_id': self.course_key})
+        save_args = {
+            'group_id': query_group,
+            'group_name': new_group_name
```
Comment (kluo) [inline lms/djangoapps/instructor/tests/test_api.py]: trailing comma

File: lms/djangoapps/instructor/tests/test_api.py [hunk 4]
```diff
@@ -422,9 +424,48 @@ def test_saving_queries(self):
...
+        response = json.loads(self.client.post(rename_url, save_args).content)
+        self.assertTrue(response["success"])
+        saved_queries = self._get_saved_queries()
+        for query in saved_queries:
+            if query['group'] == query_group:
+                self.assertEquals(query['group_title'], new_group_name)
+            else:
+                self.assertEquals(query['group_title'], u'')
+        self._delete_saved_query(query_group)
+        self.assertEquals(len(self._get_saved_queries()), 2)
```
Comment (kluo) [inline lms/djangoapps/instructor/tests/test_api.py]: wait why would this length still be 2 after you just deleted one?

File: lms/djangoapps/instructor/tests/test_api.py [hunk 5]
```diff
@@ -422,9 +424,48 @@ def test_saving_queries(self):
...
+        query_left = self._get_saved_queries()[0]
+        self._delete_saved_query(query_left['group'])
+        self.assertEquals(len(self._get_saved_queries()), 0)
+
+    def test_saving_queries_named(self):
+        """
+        Ensures we can save named grouped queries correctly
+        """
+
```
Comment (kluo) [inline lms/djangoapps/instructor/tests/test_api.py]: remove whitespace

File: lms/static/sass/course/instructor/_instructor_2.scss [hunk 6]
```diff
@@ -1009,6 +1009,37 @@ section.instructor-dashboard-content-2 {
...
+
+    .emailWidget.cancelEditName{
+      float:right;
+    }
+
+    .emailWidget.saveEditName{
+      float:left;
+    }
+
+
```
Comment (kluo) [inline lms/static/sass/course/instructor/_instructor_2.scss]: just one line

File: lms/static/sass/course/instructor/_instructor_2.scss [hunk 7]
```diff
@@ -1009,6 +1009,37 @@ section.instructor-dashboard-content-2 {
       color: $black
     }
 
+    .emailWidget.invisibleSavedGroupName{
```
Comment (kluo) [inline lms/static/sass/course/instructor/_instructor_2.scss]: add space before brace for all these

File: lms/static/sass/course/instructor/_instructor_2.scss [hunk 8]
```diff
@@ -1066,13 +1097,19 @@ section.instructor-dashboard-content-2 {
...
+      margin-right: 5px;
+    }
+
     .savequery {
       @include idashbutton($red);
       float: right;
       right: $baseline;
       font-size: 15px;
     }
-
```
Comment (kluo) [inline lms/static/sass/course/instructor/_instructor_2.scss]: why is this removed?

File: lms/static/coffee/src/instructor_dashboard/membership.coffee [hunk 9]
```diff
@@ -909,11 +911,12 @@ class EmailWidget
         created = query['created']
         type = {'text':query['type']}
         queryVals = [inclusion, type, displayEntity, filterOn]
-        invisibleTable = $('.emailWidget.invisibleQueriesStorage')
         @tr = @start_row(inclusion['text'], queryVals,
           {'class':['saved' + query.group]}, invisibleTable)
         @tr[0].setAttribute('created',created)
-        groups.add(query.group)
+        groups.add(query['group'])
+        if query['group_title'].length !=0
```
Comment (kluo) [inline lms/static/coffee/src/instructor_dashboard/membership.coffee]: space before `0`

File: lms/static/coffee/src/instructor_dashboard/membership.coffee [hunk 10]
```diff
@@ -968,18 +974,84 @@ class EmailWidget
...
+      </i> <%= label %></div>', {label: 'Rename'})
+    targ = event.target
+    while (!targ.classList.contains('editName'))
+        targ = targ.parentNode
+    targ = jQuery(targ).parent()
+    originalText = targ.text().trim().substring(6)
+    targ.html('');
+    targ.append($saveCancelEditButton)
+    htmlString = '<div class="emailWidget invisibleSavedGroupName">'+
+      originalText+'</div>'+'<input type="text" class="emailWidget editNameInput" name="queryName" value="'+originalText+'">'
```
Comment (kluo) [inline lms/static/coffee/src/instructor_dashboard/membership.coffee]: why is there an extra concatenation for `'</div>'`

**Comment by kluo** [comment]
a few comments, otherwise +1 after you fix tests

File: lms/djangoapps/instructor/tests/test_api.py (hunk 4 above)
Comment (caijim) [inline lms/djangoapps/instructor/tests/test_api.py]: 2 is the number of subqueries
